modules/programs/zsh.nix

```diff
@@ -22,7 +21,10 @@ let
     mapAttrsToList (k: v: ''hash -d ${k}="${v}"'') cfg.dirHashes
   );
 
-  zdotdir = "$HOME/" + lib.escapeShellArg cfg.dotDir;
+  # Absolute paths are assigned unmutated, relative paths
+  # are prepended with the user's home directory.
+  zdotdir = (strings.optionalString (!strings.hasPrefix "/" cfg.dotDir)
+    config.home.homeDirectory) + escapeShellArg cfg.dotDir;
```

Comment:
I had `dotDir = ".config/zsh";`  in my config and I think it generated a ZDOTDIR value of
 
```
echo $ZDOTDIR
/home/teto.config/zsh
```

Reply:
@ thanks for raising this -- looking at the code, yeah, I can see where I missed that. Curiously, it didn't cause the test to fail, which is doubly concerning -- let me made a change and test and I'll push an update :+1: